src/main/java/Helpy.java

```diff
@@ -0,0 +1,123 @@
+        System.out.println("Invalid task number. "
+                + "Please check the task number again.");
+        System.out.println(HORIZONTAL_LINE);
+    }
+
+    public static void markTask(Task task) {
+        task.setDone(true);
+
+        System.out.print(HORIZONTAL_LINE);
+        System.out.println("Good job! I've marked this task as done:");
+        System.out.println("\t[X] " + task.getTaskName());
+        System.out.println(HORIZONTAL_LINE);
+    }
+
+    public static void unmarkTask(Task task) {
+        task.setDone(false);
+
+        System.out.print(HORIZONTAL_LINE);
+        System.out.println("Ok, this task has been marked as not done yet:");
+        System.out.println("\t[ ] " + task.getTaskName());
+        System.out.println(HORIZONTAL_LINE);
+    }
+
+    public static void addTask(String command, ArrayList<Task> taskList) {
+        if (command.startsWith("mark ")) {
+            String taskNum = command.substring(5);
+            taskNum = taskNum.trim();
+            int taskIndex = Integer.parseInt(taskNum) - 1;
+            int arrayLength = taskList.toArray().length;
+
+            // Check if index is out of bounds
+            if ( taskIndex < 0 || taskIndex > arrayLength - 1) {
+                printIndexError();
+                return;
+            }
+            markTask(taskList.get(taskIndex));
+            return;
+        }
+
+        if (command.startsWith("unmark ")) {
+            String taskNum = command.substring(7);
+            taskNum = taskNum.trim();
+            int taskIndex = Integer.parseInt(taskNum) - 1;
+            int arrayLength = taskList.toArray().length;
+
+            // Check if index is out of bounds
+            if ( taskIndex < 0 || taskIndex > arrayLength - 1) {
+                printIndexError();
+                return;
+            }
+            unmarkTask(taskList.get(taskIndex));
+            return;
+        }
+
+        Task newTask = new Task(command);
+        taskList.add(newTask);
+
+        System.out.print(HORIZONTAL_LINE);
+        System.out.println("I have added: " + command);
+        System.out.println(HORIZONTAL_LINE);
+    }
+
+    public static void goodbyeUser() {
+        System.out.print(HORIZONTAL_LINE);
+        System.out.println("Goodbye, see you next time!");
+        System.out.println(HORIZONTAL_LINE);
+    }
+
+    public static void main(String[] args) {
+        greetUser();
+
+        Scanner in = new Scanner(System.in);
+        String command = "";
+        ArrayList<Task> taskList = new ArrayList<>();
+
+        while (!command.equals("bye")) {
+            command = in.nextLine();
+            switch (command) {
+            case "bye":
+                goodbyeUser();
+                break;
+            case "list":
+                listTasks(taskList);
+                break;
+            default:
+                addTask(command, taskList);
+                break;
+            }
+        }
+    }
```

Comment:
I like how clean this section of code is; Appropriate indentations and layout help to make it more readable.